### DISCUSSION TITLE
Best way to get value from disabled <input>

### DISCUSSION THREAD

**Discussion by JoaquimEsteves** [react-hook-form/react-hook-form#3157] [Q&A] [answered]
Hello,

I was wondering what was the best way to handle disabled inputs. Funnily enough there was a previous discussion that had the opposite problem to mine. https://github.com/react-hook-form/react-hook-form/discussions/2654

The previous question resolved around wanting disabled fields to not be passed to the onSubmit function, whereas I would have liked to get a hold of them for convenience's sake. 

My problem can be described as such: We fetch both form _Labels_ and _Current Data_ from some api, and construct our form programmatically depending on how the Labels are organised.

Some of these Labels, would be `noEdit` (ex: some id from the database). 

On a successful form submit I'd like to send all of those for completeness sake. The solution I found was a little... hacky and so I decided to ask if there'd be a better way.

Current Solution:

```javascript
// Assume old labels look like so:
const example_label = { foo: { noEdit: true }, bar: { required: false, regex_validation: ... } };
const example_data = { foo: "very important don't touch", bar: 'edit me' };

const Form = ({ labels, oldData }) => {
  // *snip*
  const onSubmit = async (new_data) => {
    console.log('DATA FROM ONSUBMIT 👇');
    console.log(new_data);
    console.log('AFTER SOME TWEAKS... 👇');
    const relevant_old_data = Object.keys(labels).reduce((obj, x) => {
      if (!labels[x]?.noEdit) {
        return obj;
      }
      obj[x] = oldData[x];
      return obj;
    }, {});
    const tweaked_data = { ...new_data, ...relevant_old_data };
    console.log(tweaked_data);
    // await fetch('example/api', { ...request_init, body: JSON.stringify(new_data) });
  };
};
```

**Comment by bluebill1049** [comment, marked as answer]
not sure if i am understanding your question correctly, for the input you would like to "disable"/"no edit" I think readonly is probably the attribute you want?

**Reply by JoaquimEsteves** [reply]
*Yes*! No idea why I jumped through so many hoops instead of  thinkinking about readonly.

Kind regards, and thanks for the prompt answer :)

**Reply by bluebill1049** [reply]
no worries 👍
